apps/backend/src/factory/DownloadService.ts

```diff
@@ -69,50 +70,82 @@ export async function fetchDataAndStreamResponse<TData, TMeta extends MetaBase>(
       gotResponse = true;
     });
 
-    const factoryResp = await fetch(`${ENDPOINT}/${downloadType}/${type}`, {
-      signal: controller.signal,
-      headers: {
-        Accept: 'application/json',
-        'Content-Type': 'application/json',
-      },
-      method: 'POST',
-      body: JSON.stringify(properties),
-    });
-
-    if (!factoryResp.ok) {
-      throw new Error(await factoryResp.text());
-    }
-
-    const factoryRespBody = factoryResp.body;
-    if (!factoryRespBody) {
-      return;
-    }
-
-    const readableStream = Readable.fromWeb(factoryRespBody as ReadableStream, {
-      signal: controller.signal,
-    });
-
-    readableStream.on('error', (err) => {
-      next({
-        error: err.toString(),
-        message: `Could not download generated ${downloadType}/${type}`,
-      });
-    });
-
-    const filename =
-      properties.data.length > 1
-        ? properties.meta.collectionFilename
-        : properties.meta.singleFilename;
-
-    const contentTypeHeaders = factoryResp.headers.get('content-type');
-
-    res.setHeader('Content-Disposition', contentDisposition(filename));
-    res.setHeader('x-download-filename', querystring.escape(filename));
-    if (contentTypeHeaders) {
-      res.setHeader('content-type', contentTypeHeaders);
-    }
-
-    readableStream.pipe(res);
+    const tracer = trace.getTracer('default');
+
+    await tracer.startActiveSpan(
```

Comment:
I think turning on tracing is optional we may need to use this when we have active tracing otherwise we must fall back to a basic call.